pkg/virt-launcher/virtwrap/manager.go

```diff
@@ -658,6 +659,18 @@ func (l *LibvirtDomainManager) getGuestTimeContext() context.Context {
 	return ctx
 }
 
+// FinalizeVirtualMachineMigration apply all post-migration changes on the domain.
+func (l *LibvirtDomainManager) FinalizeVirtualMachineMigration(vmi *v1.VirtualMachineInstance) error {
+	l.domainModifyLock.Lock()
```

Comment:
This PR is just moving `SetGuestTime` to be invoked from `FinalizeVirtualMachineMigration` why the extra lock is needed?

Reply:
On following commit this function also handles hostDevices attachment which change the domain so I though it will be a good idea to put a lock.
But now I realize that there is already a lock on setGuestTimeContext.
Should I remove the locks from all the function  `FinalizeVirtualMachineMigration` calls, so it will be handled from one place?
Or each function should be self contained and each function that changes the VM will have a lock?

Reply:
Ohh, I totally missed this one. From some unexplained reason I thought it was something that existed before, but it is added as a new.

A lock here is wrong, it should be placed close to where the domain if fetched and acted on.
`SetGuestTime` had no such lock before, it used a more grained lock for the timer, which looks odd and scary.

The `domainModifyLock` is needed on the "attach" operation (including the reading of the current state), so please add it there.

Reply:
Done, move the lock to `hotPlugHostDevices`